pkg/image/apis/image/types.go

```diff
@@ -51,6 +51,14 @@ const (
 	// downconverted.
 	ImporterPreferOSAnnotation = "importer.image.openshift.io/prefer-os"
 
+	// DockerImageManifestBlobLocationsAnnotation is a comma-separated list with locations of image's manifest
+	// blob and its configuration blob.
+	DockerImageManifestBlobLocationsAnnotation = "openshift.io/image.manifestBlobLocations"
```

Comment:
Do you expect values other than "registry"?

Reply:
There are three different possible locations: `registry`, `etcd` and `external`. We may want to record known manifest locations in the future (like docker daemon). The annotation would allow for encoded urls as well. Although, some new image attribute would fit better for such a purpose.

Reply:
If you plan on using URLs in the future you should be using an array of structs, JSON encoded.

Reply:
Also, constants used in apis should always be `CamelCase`

Reply:
> If you plan on using URLs in the future you should be using an array of structs, JSON encoded.

I'd rather avoid it using annotation. Let's add another attribute if the need arises in the future.

I've turned the annotation into boolean `"image.openshift.io/manifestBlobLocations"`.

> Also, constants used in apis should always be CamelCase

I may be missing something, but all the other image annotation names are in `lowerCamelCase`. On the other hand, builds [use](https://github.com/openshift/origin/blob/848776c00ea084f991a6a8b51c8f20fecc53efbc/pkg/build/apis/build/types.go#L15) `dash-separated-lower-case`.